rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/sqlite/sqlite_storage.hpp

```diff
@@ -66,14 +66,16 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
 
   uint64_t get_bagfile_size() const override;
 
+  std::string get_identifier() const override;
+
 private:
   void initialize();
   void prepare_for_writing();
   void prepare_for_reading();
   void fill_topics_and_types();
 
   std::unique_ptr<rosbag2_storage::BagMetadata> load_metadata(const std::string & uri);
-  bool database_exists(const std::string & uri);
+  static bool database_exists(const std::string & uri);
```

Comment:
I think this shouldn't have been a class member function in the first place, but that function should be a free function IMO.

Reply:
I was considering defining it as static in the cpp file, however I thought it made more sense to have it prototyped somewhere. If anyone has any other input, I'll keep it as is for now.

Reply:
it doesn't necessarily have to be a static function, can just be a free function. 
> I thought it made more sense to have it prototyped somewhere

What do you mean with this?

I just think that it makes not so much sense to have this function as a class member function.

Reply:
I wanted the definition before its use. I can define it as a free function inside `SqliteStorage.cpp`

Reply:
Just moved `database_exists` into a file-private free function inside `sqlite_storage.cpp`